internal/service/privatelinkendpoint/resource_privatelink_endpoint.go

```diff
@@ -304,21 +304,26 @@ func resourceMongoDBAtlasPrivateLinkEndpointImportState(ctx context.Context, d *
 	return []*schema.ResourceData{d}, nil
 }
 
-func resourcePrivateLinkEndpointRefreshFunc(ctx context.Context, client *matlas.Client, projectID, providerName, privateLinkID string) retry.StateRefreshFunc {
+func refreshFunc(ctx context.Context, client *admin.APIClient, projectID, providerName, privateLinkID string) retry.StateRefreshFunc {
 	return func() (any, string, error) {
-		p, resp, err := client.PrivateEndpoints.Get(ctx, projectID, providerName, privateLinkID)
+		p, resp, err := client.PrivateEndpointServicesApi.GetPrivateEndpointService(ctx, projectID, providerName, privateLinkID).Execute()
 		if err != nil {
-			if resp.Response.StatusCode == 404 {
+			if resp.StatusCode == 404 {
 				return "", "DELETED", nil
 			}
 
 			return nil, "REJECTED", err
 		}
 
-		if p.Status != "WAITING_FOR_USER" {
-			return "", p.Status, nil
+		status := ""
+		if _, ok := p.GetStatusOk(); ok {
+			status = p.GetStatus()
+		}
```

Comment:
q: why was this if added? Can see the possibility of non Ok status resulting in a "" status while this was not the case before, so just making sure no change in behavior here.

Reply:
should we return an error if the status is not ok?

Reply:
@AgustinBettati this was added because of a fmt check error:

```
Formatting go files
==> Fixing linters errors...
fieldalignment -json -fix ./...
{}
golangci-lint run --fix
internal/service/privatelinkendpoint/resource_privatelink_endpoint.go:322:13: evalOrder: may want to evaluate p.GetStatus() before the return statement (gocritic)
                return p, p.GetStatus(), nil
                          ^
make: *** [lint-fix] Error 1
```

Reply:
> should we return an error if the status is not ok?

I think returning "" should already be enough. even if you return an error (unless you mean completely failing the whole request) will end up in a retry of the request since this code is in the `refreshFunc` logic.